src/hip/hip_build_utils.cpp

```diff
@@ -51,11 +51,14 @@ static const char* MIOPEN_CMP_FLAG_REL_PATHS[MIOPEN_CMP_MAX_Flag] = {
 // Global variable List to hold the Compiler Flag Paths.
 char MIOPEN_CMP_FLAGS[MIOPEN_CMP_MAX_Flag][PATH_MAX_LEN] = {0};
 
-// Function to generate the MIOPEN Compiler Flag Value using
+// Function to generate the MIOPEN Compiler Path Value
+static char* generateCompilerPathValue(MIOPEN_CMP_Flags_t x);
+
+// Function to generate the MIOPEN Compiler Path Value using
 // ROCm Base Install Path fetched using getROCmInstallPath()
 // This approach depends on the getROCmInstallPath() provided by rocm-core
 // This flag Paths are expected to be deprecated/modified in upcoming MAJOR Releases.
-char* generateFlagValue(MIOPEN_CMP_Flags_t x)
+static char* generateCompilerPathValue(MIOPEN_CMP_Flags_t x)
```

Comment:
Pass in the string to append  and get rid of the "x" parameter..

Replace MIOPEN_CMP_FLAGS[x] with a local variable to store the result. This can be a std::string or a local array.
Replace the MIOPEN_CMP_FLAGS_REL_PATHS[x] with the parameter, the string to append.

return a strdup on the local variable holding the return value.

Get rid of the MIOPEN_CMP_FLAGS_REL_PATHS from line 48 & 49.

Reply:
Thanks for the comments, updated the changes in latest version